bin/reth/src/args/secret_key.rs

```diff
@@ -15,33 +15,46 @@ pub enum SecretKeyError {
     IOError(#[from] std::io::Error),
 }
 
-/// Attempts to load a [`SecretKey`] from a specified path. If no file exists
-/// there, then it generates a secret key and stores it in the default path. I/O
-/// errors might occur during write operations in the form of a
-/// [`SecretKeyError`]
-pub fn get_secret_key(secret_key_path: impl AsRef<Path>) -> Result<SecretKey, SecretKeyError> {
+/// Attempts to create a [`SecretKey`] at a specified path.
+/// If IO error happens while writing to secret key file, print a warning
```

Comment:
this is not the behaviour we settled on in #2498

I think this should error if creating the file fails?

Reply:
I've updated the code.